tsconfig.json

```diff
@@ -0,0 +1,32 @@
+{
+  "files": [],
```

Comment:
Eventually, the idea is that we can run tsc -b from the base folder and it will build all the projects. Of course that is limited at present by the reality of the post-build steps we need. This follows a pattern from the typescript repo itself (and is [recommended](https://www.typescriptlang.org/docs/handbook/project-references.html#overall-structure) in the docs).